src/utils/password.py

```diff
@@ -217,3 +218,120 @@ def is_in_most_used_list(self, password: str) -> bool:
         p = subprocess.Popen(command.split(), stdin=subprocess.PIPE)
         p.communicate(input=password.encode("utf-8"))
         return not bool(p.returncode)
+
+
+SECRET_KEYS = 'pass|secret|token|salt'
+SECRET_KEYS = f"{SECRET_KEYS}|{SECRET_KEYS.upper()}|pwd"
+# Avoid to match some non relevant var compound with 'key'
+SECRET_KEYS_REGEX = f"(?:\w|-)*(?:{SECRET_KEYS})\w*|local key_?| key\w*|\w(?:\w|-)*key\w*|\w*KEY\w+|\w+KEY"
+OPERATOR_REGEX = '(?:\s+\-\-value)?[\'"]?\s*(?:=|:|=>)\s*'
+VALUE_REGEX = r"'(?:[^']|'\''){5,}'|\S{5,}"
+REDACT_REGEX_STR = f"({SECRET_KEYS_REGEX}){OPERATOR_REGEX}(?:'?-----BEGIN [A-Z]+(?: '?KEY)?-----)?({VALUE_REGEX})(?:'?-----END'? [A-Z ]+-----'?)?"
+REDACT_REGEX = re.compile(REDACT_REGEX_STR)
+EXCLUDE_KEYS = [
+    "manifest_key",
+    "bind_key_",
+    "local key",
+    "local key_",
+    "version_key",
+    "version_key_",
+    "cache_key",
+    "foreign_key",
+    "primary_key",
+    "keys_zone",
+    "meta_keywords",
+    "csrf_token",
+    "jsonwebtoken",
+    "MYSQL_ROOT_PWD_FILE",
+    "SALTCORN_BIN",
+    "tls_passthrough_module",
+    "translation_key",
+    "tokenizer",
+    "teampass",
+    "misskey"
+    "calckey"
+    "libxcb-keysyms1"
+]
+EXCLUDE_KEYS_SUFFIXES = (
+    '_uri', '_url', '_path', '_key_expires', '_key_expires_date', '_enabled', '_algorithm', '_class', 'file'
+)
+EXCLUDE_VALUES = (
+    "**********",
+    "true",
+    "false",
+    "value",
+    "value1",
+    "value2",
+    "value3",
+    "version",
+    "unbound variable",
+    "db_pwd",
+    "disabled",
+    "enabled",
+    "lambda",
+    "by_order",
+    '\k\w+',
+    "(generate_random_password)",
+)
+
+def find_secrets(data: str, known_secrets: list = []) -> list:
+    for line in data.splitlines():
+        # Remove datetime and level if present
+        line = re.sub(r"^\d+-\d+-\d+ \d+:\d+:\d+,\d+: (DEBUG|ERROR|WARN(ING)?|INFO) - +", '', line)
```

Comment:
```suggestion
        line = re.sub(r"^\d+-\d+-\d+ \d+:\d+:\d+,\d+: (DEBUG|ERROR|WARN(ING)?|INFO) - ", '', line)
        line = line.strip(" +")
```

Feels like it would make sense to strip ` ` and `+` right away here (instead of later in the code which complexifies the 'if' and one may easily forget that there might be some "+" and " " remaining in there)